test/standalone/test-config-credentials.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var path = require('path');
+var assert = require('assert');
+var nock = require('nock');
+var cls = require('../../lib/cls.js');
+
+var queueSpans = function(n, privateAgent) {
+  for (var i = 0; i < n; i++) {
+    privateAgent.createRootSpanData('name', 1, 0).close();
+  }
+};
+
+process.env.GCLOUD_PROJECT = 0;
+
+describe('test-config-credentials', function() {
```

Comment:
We should have at least one test that sets both `credentials` and `keyFilename` to make sure precedence is handled correctly.